src/Microsoft.Data.SqlClient/tests/ManualTests/TracingTests/EventCounterTest.cs

```diff
@@ -16,91 +15,295 @@ namespace Microsoft.Data.SqlClient.ManualTesting.Tests
     /// </summary>
     public class EventCounterTest
     {
+        public EventCounterTest()
+        {
+            ClearConnectionPools();
+        }
+
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public void EventCounter_HardConnectionsCounters_Functional()
+        {
+            //create a non-pooled connection
+            var stringBuilder = new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString) {Pooling = false};
+
+            var ahc = SqlClientEventSourceProps.ActiveHardConnections;
+            var npc = SqlClientEventSourceProps.NonPooledConnections;
+
+            using (var conn = new SqlConnection(stringBuilder.ToString()))
+            {
+                //initially we have no open physical connections
+                Assert.Equal(ahc, SqlClientEventSourceProps.ActiveHardConnections);
+                Assert.Equal(npc, SqlClientEventSourceProps.NonPooledConnections);
```

Comment:
These assessments are not necessary. You can skip the beginning assessments for other tests too.

Reply:
Done